src/pages/instances/MigrateInstanceModal.tsx

```diff
@@ -0,0 +1,206 @@
+import { FC, KeyboardEvent, useState } from "react";
+import {
+  ActionButton,
+  Button,
+  MainTable,
+  Modal,
+} from "@canonical/react-components";
+import { LxdInstance } from "types/instance";
+import { useQuery } from "@tanstack/react-query";
+import { queryKeys } from "util/queryKeys";
+import { fetchClusterMembers } from "api/cluster";
+import Loader from "components/Loader";
+import { pluralize } from "util/instanceBulkActions";
+import { getValidInstancesForMigration } from "util/instanceMigrate";
+import ScrollableTable from "components/ScrollableTable";
+
+interface Props {
+  close: () => void;
+  migrate: (target: string, instances: LxdInstance[]) => void;
+  instances: LxdInstance[];
+}
+
+const MigrateInstanceModal: FC<Props> = ({ close, migrate, instances }) => {
+  const { data: members = [], isLoading } = useQuery({
+    queryKey: [queryKeys.cluster, queryKeys.members],
+    queryFn: fetchClusterMembers,
+  });
+
+  const [selectedMember, setSelectedMember] = useState("");
+
+  const isBulkMigration = instances.length > 1;
+
+  const handleEscKey = (e: KeyboardEvent<HTMLElement>) => {
+    if (e.key === "Escape") {
+      close();
+    }
+  };
+
+  const handleMigrate = (instances: LxdInstance[]) => {
+    migrate(selectedMember, instances);
+    close();
+  };
+
+  const handleCancel = () => {
+    if (selectedMember) {
+      setSelectedMember("");
+      return;
+    }
+
+    close();
+  };
+
+  const headers = [
+    { content: "Name", sortKey: "name" },
+    { content: "Roles", sortKey: "roles" },
+    { content: "Architecture", sortKey: "architecture" },
+    { content: "Status", sortKey: "status" },
+    { "aria-label": "Actions", className: "actions" },
+  ];
+
+  const rows = members.map((member) => {
+    const disableReason =
+      !isBulkMigration && member.server_name === instances[0].location
+        ? "Instance already running on this member"
+        : "";
+
+    const selectMember = () => {
+      if (disableReason) {
+        return;
+      }
+
+      if (selectedMember === member.server_name) {
+        setSelectedMember("");
+        return;
+      }
```

Comment:
Why is this needed, can this be removed?

Reply:
yes this is not needed indeed, will remove